echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/controllers/WebhooksController.groovy

```diff
@@ -61,11 +57,26 @@ class WebhooksController {
       }
     }
 
-    log.info("Webhook ${source}:${type}:${event.content}")
+    log.info("Webhook ${type}:${source}:${event.content}")
 
     if (sendEvent) {
       propagator.processEvent(event)
     }
   }
 
+  @RequestMapping(value = '/webhooks/{type}', method = RequestMethod.POST)
+  void forwardEvent(@PathVariable String type, @RequestBody Map postedEvent) {
+    Event event = new Event()
+    event.details = new Metadata()
+    event.details.type = type
+    event.content = postedEvent
+
+    if (event.content.source != null){
```

Comment:
> what webhook provider sends a source in it's payload?

That's a good point. I hadn't really noticed a standard for webhook payloads. If you're aware of one, I'm more than happy to adopt it. I'm really just going with the suggestions from @tomaslin and @duftler in the slack dev channel on April 7th saying that "/webhooks/teamcity" reads well, rather than something like "/webhooks/ci/teamcity". However, I also wanted to capture source if it was presented, in case one wanted to use it to trigger something else.

The plan is to add other trigger support (eg, TeamCity) but use the same code underneath for triggering.